### PR TITLE
Models added, Exceptions created, can be tested

### PR DESCRIPTION
**Pull request by NarekM**

### CODE REVIEW

File: src/com/epam/homework/university/models/Course.java [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+        return score;
+    }
+
+    public void setScore(int score) {
+        this.score = score;
+    }
+
+    public Course(String subject, int score) {
+        this.subject = subject;
+        if (score < 0 || score > 10){
```
Comment (naira-tamrazyan) [inline src/com/epam/homework/university/models/Course.java]: You may create constants for 0 and 10

File: src/com/epam/homework/university/Helper.java [hunk 2]
```diff
@@ -0,0 +1,28 @@
+package com.epam.homework.university;
+
+import com.epam.homework.university.models.Faculty;
+import com.epam.homework.university.models.Group;
+import com.epam.homework.university.models.Course;
+import com.epam.homework.university.models.Student;
+
+public class Helper {
```
Comment (naira-tamrazyan) [inline src/com/epam/homework/university/Helper.java]: You may put helper classes to a separate module like util or helper.

File: src/com/epam/homework/university/models/Faculty.java [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+        }
+        return null;
+    }
+
+    public double getAverageScoreBySubject(String subject){
+        double total = 0;
+        for (Group group : groups) {
+            total += group.getAverageScoreBySubject(subject);
+        }
+        return Double.parseDouble(new DecimalFormat("#.##").format(((double) total)/groups.length));
```
Comment (naira-tamrazyan) [inline src/com/epam/homework/university/models/Faculty.java]: the cast to double is not needed

File: src/com/epam/homework/university/models/Faculty.java [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+        return groups;
+    }
+
+    public Faculty(String name) {
+        this.name = name;
+    }
+
+    public Faculty(String name, Group[] groups) {
+        this.name = name;
+        if (groups == null || groups.length == 0){
```
Comment (naira-tamrazyan) [inline src/com/epam/homework/university/models/Faculty.java]: You may write some utility method which accepts an array and checks if it is null or empty not to repeat that check everywhere

File: src/com/epam/homework/university/models/Course.java (hunk 1 above)
Comment (NarekM) [reply src/com/epam/homework/university/models/Course.java]: Added MIN and MAX constants

File: src/com/epam/homework/university/Helper.java (hunk 2 above)
Comment (NarekM) [reply src/com/epam/homework/university/Helper.java]: Moved Helper class to helpers package

File: src/com/epam/homework/university/models/Faculty.java (hunk 3 above)
Comment (NarekM) [reply src/com/epam/homework/university/models/Faculty.java]: Removed cast to double

File: src/com/epam/homework/university/models/Faculty.java (hunk 4 above)
Comment (NarekM) [reply src/com/epam/homework/university/models/Faculty.java]: Added validateArray method to to Helper class

File: src/com/epam/homework/university/models/Course.java (hunk 1 above)
Comment (naira-tamrazyan) [reply src/com/epam/homework/university/models/Course.java]: Thanks. it is better to name it for example MIN_SCORE and MAX_SCORE to make it more understandable

**Review by naira-tamrazyan — APPROVED** [review]

File: src/com/epam/homework/university/models/Group.java [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+        Helper.validateArray(students, "Group " + name + " must have at lease one student");
+        this.students = students;
+    }
+
+    public void addStudent(Student student){
+        Helper.validateArray(students, "Group " + name + " must have at lease one student");
+        students = (Student[]) (Helper.extendArray(students, student));
+    }
+
+//    private Student[] extendArray(Student[] students, Student student){
```
Comment (naira-tamrazyan) [inline src/com/epam/homework/university/models/Group.java]: please remove the commented code before merging the code

File: src/com/epam/homework/university/models/University.java [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+        Helper.validateArray(faculties, "University "+ name +" doesn't have faculties");
+        for (Faculty faculty: faculties) {
+            if (faculty.getName().equals(name)){
+                return faculty;
+            }
+        }
+        return null;
+    }
+
+    //todo
```
Comment (naira-tamrazyan) [inline src/com/epam/homework/university/models/University.java]: todo?

File: src/com/epam/homework/university/models/Group.java (hunk 5 above)
Comment (NarekM) [reply src/com/epam/homework/university/models/Group.java]: removed

File: src/com/epam/homework/university/models/University.java (hunk 6 above)
Comment (NarekM) [reply src/com/epam/homework/university/models/University.java]: removed

**Pull request merged**